cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterController.java

```diff
@@ -83,10 +86,21 @@ public void start(Future<Void> start) {
                 configMapWatch = res.result();
 
                 log.info("Setting up periodical reconciliation for namespace {}", namespace);
-                this.reconcileTimer = vertx.setPeriodic(this.reconciliationInterval, res2 -> {
-                    log.info("Triggering periodic reconciliation for namespace {}...", namespace);
-                    reconcile();
-                });
+
+                Handler<Long> handler = new Handler<Long>() {
+
+                    @Override
+                    public void handle(Long timerId) {
+
+                        log.info("Triggering periodic reconciliation for namespace {}...", namespace);
+                        Future<?> fut = reconcile();
+                        fut.setHandler(reconcileResult -> {
+
+                            reconcileTimer = vertx.setTimer(reconciliationInterval, this);
+                        });
+                    }
+                };
+                this.reconcileTimer = vertx.setTimer(this.reconciliationInterval, handler);
```

Comment:
Do we want to start the first timer immediately? If think we used to do it and we do not do it anymore. The Watch will right after start trigger all updates and creations ... but it will not trigger any delations.